test/unittests/test_second_order_lowpass.cpp

```diff
@@ -0,0 +1,39 @@
+#include <gtest/gtest.h>
+
+#include <cmath>
+#include <path_tracking_pid/details/second_order_lowpass.hpp>
+#include <vector>
+
+using path_tracking_pid::details::SecondOrderLowpass;
+
+constexpr double eps = 1e-6;
+
+TEST(SecondOrderLowpass, StepResponse)
+{
+  double dt = 0.1;
```

Comment:
Consider making constants actually `const`. That should be your default. Only remove it, when required.

Also see https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Rconst-immutable